src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/PurposesApiServiceImpl.scala

```diff
@@ -367,4 +384,45 @@ final case class PurposesApiServiceImpl(
       }
     }
   }
+
+  override def getPurpose(purposeId: String)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerPurpose: ToEntityMarshaller[Purpose],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem]
+  ): Route = {
+    val result: Future[Purpose] = for {
+      purposeUUID       <- purposeId.toFutureUUID
+      requesterId       <- getOrganizationIdFutureUUID(contexts)
+      purpose           <- purposeProcessService.getPurpose(purposeUUID)
+      eService          <- catalogProcessService.getEServiceById(purpose.eserviceId)
+      agreement         <- getLatestAgreement(requesterId, eService, ec, agreementProcessService).flatMap(
```

Comment:
same for "use purpose consumer"
```suggestion
      purpose           <- purposeProcessService.getPurpose(purposeUUID)
      eService          <- catalogProcessService.getEServiceById(purpose.eserviceId)
      agreement         <- getLatestAgreement(purpose.consumerId, eService, ec, agreementProcessService).flatMap(
```